### DISCUSSION TITLE
[Bug]: After setting lockMovementX and lockMovementY on the object, single selection is normal, but multiple selections can be moved

### DISCUSSION THREAD

**Discussion by zhuyue6** [fabricjs/fabric.js#10317] [Q&A] [answered]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

5.3.0

### In What environments are you experiencing the problem?

Chrome

### Node Version (if applicable)

None

### Link To Reproduction

https://codesandbox.io/p/devbox/fabric-vanillajs-sandbox-forked-v7yt9f

### Steps To Reproduce

1. select two objects
2. move objects

### Expected Behavior

After multiple selections, you can’t move it

### Actual Behavior

After multiple selections, you can move it

**Comment by gloriousjob** [comment]
I'm not sure if the expected behavior is as intuitive as it sounds. Should a multiselect not be movable if one item is locked but another isn't or should it be movable because one item isn't?

You could try adding this to enforce multiselects can't move:
fabric.ActiveSelection.ownDefaults = {
	...fabric.ActiveSelection.ownDefaults,
	lockMovementX: true,
	lockMovementY: true
};

**Comment by zhe-he** [comment, marked as answer]
I think he wants to achieve this effect, and I want to achieve it too.


https://github.com/user-attachments/assets/b59a3a58-e4e2-4593-a74b-3fd86180d9d5

**Reply by alxpsr** [reply]
Could you explain how to achieve that behavior?

**Reply by alxpsr** [reply]
I tried to subscribe on `before:transform` and create new ActiveSelection with unlocked elements but it works not like in your example

**Comment by asturur** [comment]
Converted to discussion.
This has been discussed in other old issues too.
You can try to implement it with events and layout manager in the case of @zhe-he if really is a UX your product need, for the case of the multi selection you can try to use selection created or updated event to lock a multi selection that gets selected together with a locked object

**Reply by zhe-he** [reply]
I found that the "lock" mentioned by the product corresponds to the selectable property in Fabric. When an active group G contains an element A with selectable set to false, A is not affected by the rotation, scaling, translation, or flipping of G. The detailed behavior is shown in the diagram below. Since the active group has a built-in layout manager, could this effect be supported in the future?
Force the active group to include an unselectable (i.e., locked) element A, which is used solely as a placeholder.

https://github.com/user-attachments/assets/a0389c2b-a5d1-4a7b-8ef0-511e153a1c09
